python/pyarrow/_compute.pyx

```diff
@@ -2629,14 +2718,98 @@ def register_scalar_function(func, function_name, function_doc, in_types,
       21
     ]
     """
+    return register_scalar_like_function(GetRegisterScalarFunction(),
+                                         func, function_name, function_doc, in_types,
+                                         out_type, func_registry)
+
+
+def register_tabular_function(func, function_name, function_doc, in_types, out_type,
```

Comment:
I'm not sure if "tabular" is the right word. I would have expected that to be a function that takes a table/record batch and returns a new table / record batch.

What about "batch generator function"?

```suggestion
def register_batch_generator_function(func, function_name, function_doc, in_types, out_type,
```